### PR TITLE
Enforce strict mypy checks

### PR DESCRIPTION
**Pull request by derlih**
## What do these changes do?

Fix typing for mypy strict mode

## Are there changes in behavior for the user?

no

## Related issue number

#3927

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by derlih** [comment]
Locally I don't have this `mypy` errors in tests. :-(
@asvetlov @webknjaz Do we really need to run `mypy` on tests folder?

**Comment by webknjaz** [comment]
Yes. The tests are code too.

**Comment by webknjaz** [comment]
Also, can't you granularly apply the strictness to certain packages via `mypy.ini`?

**Comment by derlih** [comment]
> Yes. The tests are code too.

Well. It depends :-)
The purpose of the tests is to test correctness and functionality, but not the typing (since we using python and not c++).
I never applied type hints to tests for example. But for the code itself I usually put type hints.

**Comment by webknjaz** [comment]
Type checking can reveal problems in test just like in any other code.

File: Makefile [hunk 1]
```diff
@@ -74,7 +74,8 @@ fmt format:
 
 .PHONY: mypy
 mypy:
-	mypy --strict aiohttp tests
+	mypy --strict aiohttp
+	mypy tests
```
Comment (webknjaz) [inline Makefile]: Can't you control this via the config?

Comment (derlih) [reply Makefile]: Solved by adding pytest to lint.txt requirements file

File: CHANGES/3927.misc [hunk 2]
```diff
@@ -0,0 +1 @@
+Use mypy --strict
```
Comment (webknjaz) [inline CHANGES/3927.misc]:
```suggestion
Use ``mypy --strict``
```

File: aiohttp/web_protocol.py [hunk 3]
```diff
@@ -65,7 +65,16 @@
 ]
 
 ERROR = RawRequestMessage(
-    "UNKNOWN", "/", HttpVersion10, {}, {}, True, False, False, False, yarl.URL("/")
+    "UNKNOWN",
+    "/",
+    HttpVersion10,
+    {},  # type: ignore
+    {},  # type: ignore
```
Comment (webknjaz) [inline aiohttp/web_protocol.py]: Could you specify which errors are getting ignored? I saw syntax like `# type: ignore:SomeError` somewhere but I can't recall if it's documented.

File: CHANGES/3927.misc (hunk 2 above)
Comment (derlih) [reply CHANGES/3927.misc]: Done

File: aiohttp/web_protocol.py (hunk 3 above)
Comment (derlih) [reply aiohttp/web_protocol.py]: Done. I was not aware of this syntax.

File: aiohttp/client_reqrep.py [hunk 4]
```diff
@@ -519,10 +519,12 @@ async def write_bytes(
                 await self.body.write(writer)
             else:
                 if isinstance(self.body, (bytes, bytearray)):
-                    self.body = (self.body,)  # type: ignore
+                    body = (self.body,)
+                else:
+                    body = self.body
```
Comment (webknjaz) [inline aiohttp/client_reqrep.py]: This looks like a logic change unrelated to typing

Comment (derlih) [reply aiohttp/client_reqrep.py]: I've restored the logic

File: aiohttp/web_ws.py [hunk 5]
```diff
@@ -299,7 +309,7 @@ def ws_protocol(self) -> Optional[str]:
         return self._ws_protocol
 
     @property
-    def compress(self) -> bool:
+    def compress(self) -> int:
```
Comment (webknjaz) [inline aiohttp/web_ws.py]: Why did you change this instead of converting the return value right below?

File: tests/test_client_session.py [hunk 6]
```diff
@@ -677,8 +677,9 @@ async def test_client_session_timeout_default_args(loop: Any) -> None:
 
 
 async def test_client_session_timeout_argument() -> None:
-    session = ClientSession(timeout=500)
-    assert session.timeout == 500
+    timeout = ClientTimeout(total=500)
+    session = ClientSession(timeout=timeout)
```
Comment (webknjaz) [inline tests/test_client_session.py]: Why can't session timeout accept `int`?

**Comment by derlih** [comment]
@webknjaz I removed all changes and specified only mypy error codes in `type: ignore[code]`

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
@derlih mind backporting this to 3.8? https://docs.aiohttp.org/en/stable/contributing.html#backporting

**Comment by derlih** [comment]
> @derlih mind backporting this to 3.8? https://docs.aiohttp.org/en/stable/contributing.html#backporting

I will try to
